extensions/amp-scroll-toggle/validator-amp-scroll-toggle.protoascii

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+tags: {  # amp-scroll-toggle
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-scroll-toggle"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-scroll-toggle>
+  html_format: AMP
+  tag_name: "AMP-SCROLL-TOGGLE"
+  requires_extension: "amp-scroll-toggle"
+  attrs: {
+    name: "position"
```

Comment:
just curious, did we consider implementing this as an attribute instead of a tag?

Reply:
That's an interesting idea. It could be less verbose. @aghassemi for thoughts.

It would look like this:

- `toggle-on-scroll` can be set on any element (probably need to extend a default attr list?) and defines position with optional value. default is `top`. `media` on the target itself already takes care of all media query responsibilities.
- The `...Dispatcher` service scans when the doc is ready. Maybe also when the DOM is updated.

Reply:
I like the attribute idea. how about we put it in `amp-fx-collection`? fits nicely there and we already have all the scanning mechanisms. `amp-fx-collection` uses `data-<fx-name>` for configuration.

```
<header amp-fx="scroll-toggle" data-scroll-toggle-position="top">
```

`media` support for all of amp-fx-collection would be nice anyway

Reply:
SG. Debating name, since `scroll-toggle` is a little confusing in the FX context since other effects can be understood to be "scroll-bound".

I was thinking maybe name this `float-in` for consistency and differentiation with `fly-in-*`?

(Also note that `amp-fx-collection` gives us an overhead of ~4.8K, up to you whether that's an issue.)

Reply:
I don't have a strong opinion whether separate extension or `fx-collection`. Too many extensions vs properly grouped extensions has its own trade-offs. There is no strong signal IMO whether someone who wants this also needs other stuff in `fx-collection` (whereas arguably someone who needs `fly-in` may also want `fade-in`). I am fine either way.

Reply:
(f you go with fx-collection, maybe `float-top`, `float-bottom` )

Reply:
tangent: JS files that are less than 1kb do not get byte code cached. if the new extension ends up say 900B, it may be worth to actually increase its size.

Reply:
`amp-scroll-toggle` is `2.6K` gzipped by itself.
`amp-fx-collection` with scroll-toggle stuff is `5.8K`.

I guess since the delta (`~1K`) is smaller than the `amp-scroll-toggle` by itself maybe it's worth adding to `fx-collection` after all. A roundtrip is a roundtrip.